### DISCUSSION TITLE
Need to add /src after the import statement to be able to use type definitions

### DISCUSSION THREAD

**Discussion by RubenPauwelsUniWeb** [callstack/react-native-builder-bob#507] [Q&A]
We are using a generated tarball, using the default config.

Our build command looks like this:
`yarn prepack && npm pack`

When we try to import the package in our project (after adding it with `yarn add`) we need to specify the import statement like this: `import {useCustomHook} from 'our-package/src';`

If we don't add `/src` to the import we get this TS error: `Cannot find module 'our-package' or its corresponding type declarations.ts(2307)`

Is it possible to omit the `/src` and still be able to use the type definitions?

**Comment by satya164** [comment]
Adding `/src` shouldn't be necessary. Are your type definitions properly generated under `lib/typescript` and does the `types` entry in `package.json` point to the correct file?

**Comment by RubenPauwelsUniWeb** [comment]
![image](https://github.com/callstack/react-native-builder-bob/assets/122032001/8696f1cc-e5c8-4f40-a60f-08e955b93f10)
![image](https://github.com/callstack/react-native-builder-bob/assets/122032001/2e553c09-f5cf-437a-bdcd-8169d7ed65c6)


As you can see they are there. For some reason they are in `lib/typescript/src` instead of `lib/typescript`.
I also tried changing the types property in package.json to `"types": "lib/typescript/src/index.d.ts"` but this gives the same issue and it's still only working when adding `/src` to the import.

**Reply by satya164** [reply]
You need to have `lib/typescript/src/index.d.ts` for your `types` field. If you're using latest version of bob `bob build` would've thrown an error due to the wrong path.

As per why it doesn't work, can't say without more without code.
